pyccel/codegen/compiling/compilers.py

```diff
@@ -441,7 +441,10 @@ def compile_shared_library(self, compile_obj, output_folder, verbose = False, sh
         # Collect compile information
         exec_cmd, includes, libs_flags, libdirs_flags, m_code = \
                 self._get_compile_components(compile_obj, accelerators)
-        linker_libdirs_flags = ['-Wl,-rpath' if l == '-L' else l for l in libdirs_flags]
+        if self._info['exec'] in ('nvcc', 'nvc', 'nvfortran'):
+            linker_libdirs_flags = ['-Xcompiler' if l == '-L' else f'"-Wl,-rpath,{l}"' for l in libdirs_flags]
```

Comment:
Did you ever get to the bottom of why this code wasn't being called? Is it called correctly now?

Are you sure that this does the same job as the original expression?

Reply:
[6e7ae80](https://github.com/pyccel/pyccel-cuda/pull/32/commits/6e7ae8068b7b218213802be39a80d1159476e20c)